### PR TITLE
Align behaviour of `AttributeDict` and `AttributeList`

### PR DESCRIPTION
**Pull request by maratori**
`AttributeList.remove()` didn't raise exception if `attribute` not found in list. However `AttributeDict.pop()` did raise exception if `name` not found in dict.
Now they behave the same - no exception.

### CODE REVIEW

File: lona/html/attribute_dict.py [hunk 1]
```diff
@@ -19,6 +19,9 @@ def items(self):
 
     def pop(self, name):
         with self._node.lock:
+            if name not in self._attributes:
+                return None
```
Comment (fscherf) [inline lona/html/attribute_dict.py:23]: Does this make sense? The reason why this method raises a key error when the given name is not present in ``_attributes`` is that ``.pop()`` is supposed to to return the value of ``name`` and ``.remove()`` does not do this.

File: lona/html/data_binding/inputs.py [hunk 2]
```diff
@@ -38,13 +38,14 @@ def handle_change(self, input_event):
         # The solution for this problem is to don't send patches back to users
         # who issued them.
 
-        self.attributes.__setitem__(
-            'value',
-            input_event.data,
-            issuer=(input_event.connection, input_event.window_id),
-        )
+        with self.lock:
```
Comment (fscherf) [inline lona/html/data_binding/inputs.py:41]: Why is this necessary?  ``AttributeDict.__setitem__()`` pulls the lock for us.

File: lona/html/node.py [hunk 3]
```diff
@@ -177,11 +177,8 @@ def ignore(self):
     def ignore(self, value):
         if value:
             self._attributes['data-lona-ignore'] = 'True'
-
```
Comment (fscherf) [inline lona/html/node.py:180]: I want to keep this newline to keep styling consistent with the rest of the code

**Review by fscherf — CHANGES_REQUESTED** [review]
Great changes in general :+1:  I am ok with everything i didn't comment on

File: lona/html/attribute_dict.py (hunk 1 above)
Comment (maratori) [reply lona/html/attribute_dict.py:23]: What do you think if I change it as following?
1. `attributes.pop(name)` will raise an exception if name is unknown
2. `attributes.pop(name, default)` will not raise an exception, but return `default`
3. `del attributes[name]` will not raise an exception

File: lona/html/data_binding/inputs.py (hunk 2 above)
Comment (maratori) [reply lona/html/data_binding/inputs.py:41]: Lock may be dropped if we can drop we can drop `self.value = input_event.data` and use only `self.attributes.__setitem__`.
Otherwise there is a data race between them.

But you are right, that this change should be in a separate PR.

**Comment by fscherf** [comment]
@maratori oh ok. big pr's with smaller commits are also fine
